src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs

```diff
@@ -0,0 +1,162 @@
+        }
+
+        [Theory]
+        [InlineData(typeof(MulticastDelegate))]
+        [InlineData(typeof(OtherNonGenericDelegate))]
+        public void GetDelegateForFunctionPointer_InvalidType_ReturnsExpected(Type type)
+        {
+            MethodInfo targetMethod = typeof(GetDelegateForFunctionPointerTests).GetMethod(nameof(Method));
+            Delegate d = targetMethod.CreateDelegate(typeof(NonGenericDelegate));
+            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(d);
+
+            Delegate functionDelegate = Assert.IsType<NonGenericDelegate>(Marshal.GetDelegateForFunctionPointer(ptr, type));
+            Assert.NotNull(functionDelegate);
+            Assert.IsType<NonGenericDelegate>(functionDelegate);
+            Assert.Equal(targetMethod, functionDelegate.Method);
+            Assert.Null(functionDelegate.Target);
+        }
+
+        [Fact]
+        public void GetDelegateForFunctionPointer_CollectibleType_ReturnsExpected()
+        {
+            MethodInfo targetMethod = typeof(GetDelegateForFunctionPointerTests).GetMethod(nameof(Method));
+            Delegate d = targetMethod.CreateDelegate(typeof(NonGenericDelegate));
+            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(d);
+
+            AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("Assembly"), AssemblyBuilderAccess.RunAndCollect);
+            ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule("Module");
+            TypeBuilder typeBuilder = moduleBuilder.DefineType("Type", TypeAttributes.Class | TypeAttributes.Public | TypeAttributes.Sealed | TypeAttributes.AnsiClass | TypeAttributes.AutoClass, typeof(MulticastDelegate));
+            ConstructorBuilder constructorBuilder = typeBuilder.DefineConstructor(MethodAttributes.RTSpecialName | MethodAttributes.HideBySig | MethodAttributes.Public, CallingConventions.Standard, new Type[] { typeof(object), typeof(IntPtr) });
+            constructorBuilder.SetImplementationFlags(MethodImplAttributes.Runtime | MethodImplAttributes.Managed);
+
+            MethodBuilder methodBuilder = typeBuilder.DefineMethod("Invoke", MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.NewSlot | MethodAttributes.Virtual, targetMethod.ReturnType, targetMethod.GetParameters().Select(p => p.ParameterType).ToArray());
+            methodBuilder.SetImplementationFlags(MethodImplAttributes.Runtime | MethodImplAttributes.Managed);
+
+            Type type = typeBuilder.CreateType();
+
+            Delegate functionDelegate = Marshal.GetDelegateForFunctionPointer(ptr, type);
+            Assert.NotNull(functionDelegate);
+            Assert.IsType<NonGenericDelegate>(functionDelegate);
+            Assert.Equal(targetMethod, functionDelegate.Method);
+            Assert.Null(functionDelegate.Target);
+        }
+
+        [Fact]
+        public void GetDelegateForFunctionPointer_Generic_ReturnsExpected()
+        {
```

Comment:
whts the difference between this test and ```GetDelegateForFunctionPointer_NonGeneric_ReturnsExpected``` ?

Reply:
Fixed